cabbage/postgres.py

```diff
@@ -1,39 +1,85 @@
-import functools
+from typing import Any, Dict, Iterator, Union
+
 import psycopg2
+from psycopg2 import extras, sql
+from psycopg2.extras import RealDictCursor
+
+from cabbage import exceptions, types
+
 
-from psycopg2 import sql
+def init_pg_extensions() -> None:
+    psycopg2.extensions.register_adapter(dict, extras.Json)
 
 
-def launch_task(queue, name, lock, kwargs):
+def launch_task(queue: str, name: str, lock: str, kwargs: types.JSONValue) -> int:
 
     conn = get_global_connection()
     with conn.cursor() as cursor:
         cursor.execute(
             """INSERT INTO tasks (queue_id, task_type, targeted_object, args)
-               SELECT id, %s, %s, %s FROM queues WHERE queue_name=%s;""",
+               SELECT id, %s, %s, %s FROM queues WHERE queue_name=%s
+               RETURNING id;""",
             (name, lock, kwargs, queue),
         )
+        row = cursor.fetchone()
+
+    if not row:
+        raise exceptions.QueueNotFound(queue)
+
     conn.commit()
+    return row[0]
+
+
+def get_tasks(cursor: Any, queue: str) -> Iterator[Dict[str, Union[str, int]]]:
```

Comment:
return a namedtupled instead of dict : it will be easier to handle typing checks